doc/design_decisions.rst

```diff
@@ -0,0 +1,87 @@
+Design decisions
+================
+
+
+Single Pool instance
+--------------------
+
+We have decided to keep all packages incl. comps in a single Pool instance.
+Lookups are slower due to iterating through a larger data set
+(using smaller dedicate Pool instances per package type would be faster),
+but we don't have to duplicate repo definitions, keep the repos in sync
+across all Pool instances etc.
+
+Modularity implementation is an exception to this rule
+because the modulemd records require special treatment.
+
+
+Working with multiple package types
+-----------------------------------
+
+It is possible to load multiple package types (such as rpm, deb, arch, ...)
+in a single libsolv Pool and make them available via PackageSack.
+We recommend doing so only for querying purposes.
+We MUST avoid running a transaction across multiple package types
+because the respective databases (rpmdb, dpkg database, ...) would contain
+partial records with broken dependencies and would be impossible to manage.
+
+
+System state vs history database
+--------------------------------
+
+System state and history database hold similar data, but they are not interchangeable.
+That's why we must use them for the right purposes.
+
+* System state
+
+  * Holds information about the currently installed software (extends rpmdb and any other state information).
+  * Is used to retrieve information about installed software such as
+    ``reason``, ``from_repo``, list of installed comps groups and their packages etc.
+
+* History database
+
+  * Holds recorded transactions.
+  * Is used in ``history`` operations to rollback/undo/redo transactions.
```

Comment:
... and then here, one more purpose for the history is auditing, but another bullet point: It is not crucial for maintaining the installed system and may be freely deleted.

Reply:
Updated. I just used ``managing`` instead of ``maintaining``.